src/Knp/Menu/Integration/Pimple/PimpleMenuProvider.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Knp\Menu\Integration\Pimple;
+
+use Knp\Menu\Provider\MenuProviderInterface;
+
+class PimpleMenuProvider implements MenuProviderInterface
+{
+    private $pimple;
+    private $menuIds;
+
+    public function __construct(\Pimple $pimple, array $menuIds = array())
```

Comment:
supporting both Pimple 1 and Pimple 3 would be easy, as we just read from Pimple here: just remove the typehint here and use a instanceof-based check in the constructor allowing for both class names.

Supporting Silex 1.x and 2.x will be much harder (and IMO, it would be a good idea to split the Silex provider to a separate package, just like the symfony bundle is separate. Much easier to support multiple Silex version as we can have an independent versionning)